### PR TITLE
568175: Ingestion Worker: Add a cutoff to prevent super-large families

### PR DESCRIPTION
**Pull request by andyreidz**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=568175

### CODE REVIEW

**Comment by andyreidz** [comment]
@dermot-hardy for a first cut I went with a sub document limit, byte limit looks a bit harder but everything is doable.

**Comment by andyreidz** [comment]
Sorry, I forgot to address the exceptions, doing it now.

**Comment by andyreidz** [comment]
Should be good to go now.

File: worker-batch-ingestion-plugin/src/test/java/com/github/cafdataprocessing/worker/ingestion/DocumentWorkerDocumentDeserializerTest.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        final String json = "{\"subdocuments\":{}}";
+
+        try {
+            final DocumentWorkerDocument documentWorkerDocument =
+                    objectMapper.readValue(json, DocumentWorkerDocument.class);
+        }
+        catch(final IllegalStateException ex) {
+            assertEquals("Expected '[' at [Source: (String)\"{\"subdocuments\":{}}\"; line: 1, column: 18]", ex.getMessage());
+        }
+
```
Comment (frankmccarry) [inline worker-batch-ingestion-plugin/src/test/java/com/github/cafdataprocessing/worker/ingestion/DocumentWorkerDocumentDeserializerTest.java:120]: Looks good, I've no issues with it other than it passes if the Json Exception is thrown and not tested for in the test and local variable not used. I'd write it this way to only allow test to pass if IllegalState thrown
    @Test
    void deserializeInvalidSubdocumentsTest() {

        final String json = "{\"subdocuments\":{}}";
        IllegalStateException thrown = Assertions.assertThrows(IllegalStateException.class, () -> {
            objectMapper.readValue(json, DocumentWorkerDocument.class);
        });
        Assertions.assertEquals("Expected '[' at [Source: (String)\"{\"subdocuments\":{}}\"; line: 1, column: 18]",
                thrown.getMessage());
    }

**Review by frankmccarry — APPROVED** [review]
Looks good

File: worker-batch-ingestion-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Ingestion Batch Worker Plugin
+
+## Configuration
+
+### Environment Variables
+
+- **CAF_STAGING_SERVICE_BASEPATH**
+  - The staging service base path, required.
```
Comment (michael-bryson) [inline worker-batch-ingestion-plugin/README.md:8]: I see this says required, is this newly required or did it already exist? If it is new and is required I would think this constitutes a breaking change.

Comment (andyreidz) [reply worker-batch-ingestion-plugin/README.md:8]: It already existed, since I was adding doc I thought it should be mentioned.

Comment (andyreidz) [reply worker-batch-ingestion-plugin/README.md:8]: image

**Review by michael-bryson — APPROVED** [review]

**Pull request merged**